test/test-udp-recvmsg-unreachable-error.c

```diff
@@ -0,0 +1,125 @@
+/* Copyright libuv project and contributors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "task.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#define CLIENT_TEST_PORT 9123
+#define SERVER_TEST_PORT 9124
+
+int recv_cb_called = 0;
+
+void udp_send_cb(uv_udp_send_t* req, int status) {
+  ASSERT_EQ(status, 0);
+}
+
+void alloc_cb(uv_handle_t* handle,
+              size_t suggested_size,
+              uv_buf_t* buf) {
+  *buf = uv_buf_init(malloc(suggested_size), suggested_size);
```

Comment:
Don't ignore malloc allocation errors. In tests we commonly use a static buffer instead, that way you don't have to free it afterwards either.